### DISCUSSION TITLE
Getting JMX metrics going with Bitnami Kafka

### DISCUSSION THREAD

**Discussion by markbrock301** [provectus/kafka-ui#3360] [Q&A] [answered]
Hi team,

I really like the UI you have built, I was wondering though if you could offer some advice, or better still a working example, in getting the metrics feature to work with the Bitnami Kafka helm distribution.  

All of your examples demonstrating metrics appear to utilize Confluent Kafka, which uses KAFKA_JFX_OPTS and JFX_PORT to configure a metrics endpoint.  The Bitnami helm chart has a bunch of fields to stand up JFX and Kafka exporters, which I have enabled and pointed the UI at, but I am not clear if this doing the same thing as Confluent Kafka in your examples (i.e. see metrics.jmx.* and metrics.kafka.* parameters on https://github.com/bitnami/charts/tree/main/bitnami/kafka, and refer to https://docs.bitnami.com/kubernetes/infrastructure/kafka/administration/enable-metrics/)

Ultimately the Kafka UI pod is throwing an error saying it can't connect to the JSX metrics endpoint.   

It's just the metrics I am having trouble with, everything else seems to work well.

Appreciate any guidance you might have.

Thanks,
Mark

**Comment by Haarolean** [comment, marked as answer]
Hi,

considering the documentation you provided, there are either (I don't understand clearly) two types of metrics available for bitnami (jmx and prometheus) or even one -- prometheus. Keep in mind that we use JMX as a default interface.

Here's a [compose file](https://github.com/provectus/kafka-ui/blob/master/documentation/compose/kafka-ui-with-jmx-exporter.yaml) with kafka-ui set to use prometheus.

Also, take a look here:
https://stackoverflow.com/questions/73083028/how-to-configure-kafka-chart-to-expose-both-kafka-metrics-and-kafka-jmx-metrics
https://hub.docker.com/r/bitnami/jmx-exporter/
